src/options.c

```diff
@@ -81,6 +83,44 @@ static int parse_ipv4(struct super_block *sb, char *str,
 	return 0;
 }
 
+static int parse_metadev_path(struct super_block *sb, substring_t *substr,
+			      char **md_path)
+{
+	char *metadev_path;
+	struct inode *metadev_inode;
+	struct path path;
+	int error;
+
+	/* 1. check metadev is a block device */
+	metadev_path = match_strdup(substr);
+	if (!metadev_path) {
+		scoutfs_err(sb, "metadev string dup failed");
+		return -ENOMEM;
+	}
+
+	error = kern_path(metadev_path, LOOKUP_FOLLOW, &path);
+	if (error) {
+		scoutfs_err(sb, "path %s not found for metadev: error %d",
+			    metadev_path, error);
+		kfree(metadev_path);
+		return error;
+	}
+
+	metadev_inode = d_inode(path.dentry);
+	if (!S_ISBLK(metadev_inode->i_mode)) {
+		scoutfs_err(sb, "path %s for metadev is not a block device",
+			    metadev_path);
+		path_put(&path);
```

Comment:
There's enough cases of frees and puts that we should move to having one exit path that cleans up.  Free the path if we're returning an error, otherwise give it to the caller.  Have a bool that tracks if we got the path, and test that for putting it (annoying that the api doesn't support initializing the path and always calling put).  All the error paths goto out instead of returning mid-function.  Probably change error to ret?